editor/components/block-list/block.js

```diff
@@ -650,10 +649,21 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps ) => {
 		onSelect( uid = ownProps.uid, initialPosition ) {
 			selectBlock( uid, initialPosition );
 		},
-		onInsertBlocks( blocks, index ) {
-			const { rootUID, layout } = ownProps;
-			blocks = blocks.map( ( block ) => cloneBlock( block, { layout } ) );
-			insertBlocks( blocks, index, rootUID );
+		onInsertBlocksAfter( blocks ) {
+			const { block, order, isLast, rootUID, orderOfRoot, rootUIDOfRoot, layout } = ownProps;
+
+			blocks = blocks.map( ( oldBlock ) => cloneBlock( oldBlock, { layout } ) );
+
+			if (
+				rootUID && isLast && blocks.length === 1 &&
+				isUnmodifiedDefaultBlock( first( blocks ) ) &&
+				isUnmodifiedDefaultBlock( block )
+			) {
+				removeBlock( block.uid );
+				insertBlocks( blocks, orderOfRoot + 1, rootUIDOfRoot );
+			} else {
+				insertBlocks( blocks, order + 1, rootUID );
+			}
```

Comment:
Can you explain the need of this separate handler and its logic, not obvious at first sight :)

Reply:
Oh sure! Forgot to comment. It's so escape the nested block list on double enter.

Reply:
A small e2e test for the double enter behavior might be good. It's something that can regress easily without noticing.